.github/workflows/sync.yml

```diff
@@ -0,0 +1,52 @@
+name: Sync
+
+on:
+  push:
+    branches: [master, staging]
+  workflow_dispatch:
+
+jobs:
```

Comment:
I think we want to make sure all our CI checks run successfully before deploying to avoid shipping broken code to production. Since it seems like there's no way to make one workflow depend on another and your `on` events are almost the same, maybe we should move this into the main `ci-cd.yml` file with `needs` for dependencies and some more `if`s. https://github.com/CruGlobal/missionhub-react-native/blob/develop/.github/workflows/ci-cd.yml has some exaples of `needs` depending on other job successes and `if` checking for branches and non PRs (just push events).